fineract-provider/src/main/java/org/apache/fineract/commands/service/IdempotencyKeyGenerator.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.commands.service;
+
+import java.util.UUID;
+
+public final class IdempotencyKeyGenerator {
+
+    private IdempotencyKeyGenerator() {}
+
+    public static String create() {
+        return UUID.randomUUID().toString();
```

Comment:
Big no-no. This definitely shouldn't be a static factory for the idempotency keys but a Spring bean. Makes future testing easier + if you make an interface behind the bean class, it can be replaced with the existing modularization approach so one could alter the idempotency key generation.

Reply:
done

Reply:
done